src/agent/onefuzz-agent/src/tasks/config.rs

```diff
@@ -79,9 +81,16 @@ pub enum Config {
 }
 
 impl Config {
-    pub fn from_file(path: impl AsRef<Path>) -> Result<Self> {
-        let json = std::fs::read_to_string(path)?;
-        Ok(serde_json::from_str(&json)?)
+    pub fn from_file(path: impl AsRef<Path>, setup_dir: Option<impl AsRef<Path>>) -> Result<Self> {
+        let file = std::fs::File::open(path)?;
+        let reader = std::io::BufReader::new(file);
+        let mut json_config: serde_json::Value = serde_json::from_reader(reader)?;
+        if let Some(setup_dir) = setup_dir {
```

Comment:
A few things.  This appears to be optional, but the CommonConfig entry is not.

Since this uses the command line arguments, instead of inserting into the json prior to deserialization, why not make setup_dir optional, de-serialize the json as is, then set the value to the object?

This could be replaced with:

```
let mut config: Self = serde_json::from_value(json_config)?;
config.setup_dir = setup_dir;
```

Reply:
The idea is to fail fast if the setup_dir is not specified in the config or the command line. I wanted to avoid the situation where the setup_dir not specified and we only know it while the task is running.

Reply:
IIRC, this is a convenience for local fuzzing users, so they don't _have_ to pass the `setup_dir` as an arg. On a remote node, the `setup_dir` argument to this function will always be present.

I'm fine with removing this convenience, and making this `setup_dir` function arg non-optional too.

My thinking is that in the future, we want to move to a more unified description of all FS-local contextual info (local paths, abstract work queue "provider", &c), and that will subsume any current pain in the local use case.

Reply:
Whatever we end up with here, I do think we should add  a doc comment for this `from_file()` ctor, explaining what's going on with the "stomped in" value.